### PR TITLE
User Signup Form Validation and Toast Message

### LINKED ISSUE #654: [Onboarding Flow]- User Signup Form Validation and Toast Message
# User Signup Form Validation and Toast Message
## Overview:

We need to implement user input validation for the signup form. Specifically, we want to ensure that the input fields for "firstname" and "lastname" meet certain criteria, and if the criteria are not met, display an error message. Additionally, when user is sign out and trying to generate username
should trigger a toast message instructing the user to sign in.

## User Stories:

1. As a user, I want the input fields for "firstname" and "lastname" to:
   - any input not contains spaces.
   - any input not contains non-alphabetical characters.
   - any input not contains numbers.
   - any input not longer than 20 characters.
   - If the input in the "firstname" or "lastname" fields violates any of the above criteria, then display an error message below the input field firstname and lastname if any of the above criteria are not met.
   - If the input in the "firstname" and "lastname" fields meet the criteria, a username should be generated.

2. As a user, when I click the "generateUsername" button and the "sign-in-with-github" session is logged out:
    - A toast message should be displayed, saying "Please sign in."

## Requirement:

Wireframe: https://excalidraw.com/#json=aknBUT8kGMRp2GLESYGcp,NNAIFlOSSeVt-eLbyv51Fw

image

    
## Acceptance Criteria:

- [x] TDD
- [x] Allow user to input anything in "firstname" and "lastname" input field
- [x] If the input in the "firstname" or "lastname" fields violates certain criteria i.e. input not contain spaces, non-alphabetical characters, numbers, more than 20 character, an error message should be displayed below the respective input field ("firstname" and "lastname") and generateusername button will disable
- [x] If the input in both the "firstname" and "lastname" fields meet the criteria, a button will enable and a username should be generated.
- [x] When the "sign-in-with-github" session is logged out, and the "generateUsername" button is clicked
- [x]  A toast message should be displayed with the text "Please sign in."
- [x] feature under feature flag

### PR DESCRIPTION
**Pull request by shubhamsigdar1**
### Issue Closes #654 

Parent ticket: https://github.com/Real-Dev-Squad/todo-action-items/issues/178

### What is the change?

- Allow user to input anything in "firstname" and "lastname" input field

- If the input in the "firstname" or "lastname" fields violates certain criteria i.e. input not contain spaces, non-alphabetical characters, numbers, more than 20 character, an error message should be displayed below the respective input field ("firstname" and "lastname") and generateusername button will disable
-  If the input in both the "firstname" and "lastname" fields meet the criteria, a button will enable and a username should be generated.
- When the "sign-in-with-github" session is logged out, and the "generateUsername" button is clicked, A toast message should be displayed.
- Feature under feature flag

### Is Development Tested?

- [X] Yes
- [ ] No

### After Change Video

https://github.com/Real-Dev-Squad/website-www/assets/107163260/a21b8dda-44e4-4d54-b322-9f1699c71ad6

Test video:

https://github.com/Real-Dev-Squad/website-www/assets/107163260/b7add596-f1b9-4e9e-93b7-f87fe7974856

### CODE REVIEW

File: app/components/signup-steps/step-one.js [hunk 1]
```diff
@@ -54,8 +87,16 @@ export default class SignupStepsStepOneComponent extends Component {
...
       );
       const data = await response.json();
-      this.username = data.username;
-      this.args.setUsername(this.username);
+      if (response.status === 200) {
+        this.username = data.username;
+        this.args.setUsername(this.username);
+      } else if (response.status === 401) {
+        this.toast.error(
+          'You are not logged in. Please login to continue.',
```
Comment (iamitprakash) [inline app/components/signup-steps/step-one.js:95]: change text to "Please login to continue."

File: app/components/signup-steps/step-one.js [hunk 2]
```diff
@@ -54,8 +87,16 @@ export default class SignupStepsStepOneComponent extends Component {
...
+          '',
+          toastNotificationTimeoutOptions
+        );
+      }
     } catch (err) {
       console.log('Error: ', err);
```
Comment (iamitprakash) [inline app/components/signup-steps/step-one.js:101]: replace err to "Something went wrong"

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: app/components/signup-steps/step-one.js (hunk 1 above)
Comment (shubhamsigdar1) [reply app/components/signup-steps/step-one.js]: Done with changes...can you please check my latest commit

File: app/components/signup-steps/step-one.js (hunk 2 above)
Comment (shubhamsigdar1) [reply app/components/signup-steps/step-one.js]: Done with changes...can you please check my latest commit

**Review by iamitprakash — APPROVED** [review]

**Review by vinit717 — APPROVED** [review]

**Pull request merged**